### PR TITLE
feat(docs): Add clustered clickhouse docs to self-hosted config

### PR DESCRIPTION
**Pull request by angus-langchain**

### CODE REVIEW

**Review by agola11 — APPROVED** [review]

File: docs/self_hosting/configuration/external_clickhouse.mdx [hunk 1]
```diff
@@ -79,9 +81,19 @@ CLICKHOUSE_PORT=8123 # Change to your Clickhouse port if needed
...
 
 Once configured, you should be able to reinstall your LangSmith instance. If everything is configured correctly, your LangSmith instance should now be using your external ClickHouse database.
+
+:::note
+When using a clustered deployment, LangSmith will automatically:
+
+- Run database migrations across all nodes in the cluster
+- Configure tables for data replication across the cluster
+
+Note that while data is replicated across nodes, LangSmith does not configure distributed tables or handle query routing - queries will be directed to the specified host. You will need to handle any load balancing or query distribution at the infrastructure level if desired.
```
Comment (akira) [inline docs/self_hosting/configuration/external_clickhouse.mdx:98]: Should we mention that you cannot enable this on an existing cluster?

Comment (akira) [inline docs/self_hosting/configuration/external_clickhouse.mdx:98]: Should we mention that you cannot enable this on an existing cluster?

File: docs/self_hosting/configuration/external_clickhouse.mdx [hunk 2]
```diff
@@ -79,9 +81,19 @@ CLICKHOUSE_PORT=8123 # Change to your Clickhouse port if needed
...
+CLICKHOUSE_CLUSTER=my_cluster_name # Optional: Set this if using a clustered deployment
 `
     ),
   ]}
 />
 
 Once configured, you should be able to reinstall your LangSmith instance. If everything is configured correctly, your LangSmith instance should now be using your external ClickHouse database.
+
+:::note
+When using a clustered deployment, LangSmith will automatically:
```
Comment (akira) [inline docs/self_hosting/configuration/external_clickhouse.mdx:93]: Should we mention this only works with an "external" clickhouse?  Right now we don't add the keeper configurations, so wouldn't work with our k8s setup (@langchain-infra if any specific comments here).

**Review by langchain-infra — APPROVED** [review]

File: docs/self_hosting/configuration/external_clickhouse.mdx [hunk 3]
```diff
@@ -79,9 +81,22 @@ CLICKHOUSE_PORT=8123 # Change to your Clickhouse port if needed
...
   ]}
 />
 
 Once configured, you should be able to reinstall your LangSmith instance. If everything is configured correctly, your LangSmith instance should now be using your external ClickHouse database.
+
+:::note
+Important considerations for clustered deployments:
+
+- Clustered setups must be configured on a fresh schema - existing standalone ClickHouse instances cannot be converted to clustered mode.
+- Clustering is only supported with externally managed ClickHouse deployments. It is not compatible with LangSmith-managed ClickHouse installations as these do not include required ZooKeeper configurations.
```
Comment (akira) [inline docs/self_hosting/configuration/external_clickhouse.mdx:96]: `It is not compatible with self hosted Clickhouse installations installed through the LangSmith Helm, as these do not include required Keeper configurations.`

File: docs/self_hosting/configuration/external_clickhouse.mdx [hunk 4]
```diff
@@ -33,7 +33,7 @@ Additionally, sensitive information can be configured to be not stored in Clickh
 
 - A provisioned ClickHouse instance that your LangSmith application will have network access to (see above for options).
 - A user with admin access to the ClickHouse database. This user will be used to create the necessary tables, indexes, and views.
-- We only support standalone ClickHouse (not clustered or replicated) or ClickHouse Cloud.
+- We support both standalone ClickHouse and externally managed clustered deployments. For clustered deployments, ensure all nodes are running the same version. Note that clustered setups are not supported with LangSmith-managed ClickHouse installations.
```
Comment (akira) [inline docs/self_hosting/configuration/external_clickhouse.mdx:36]: Same here as below on "not supported". 

For LangSmith-managed ClickHouse installations you can say:  Note that Langsmith-managed Clickhouse installations are automatically clustered as they are using SharedMerge, which offers separation of compute and storage.

**Pull request merged**